openedx/features/course_experience/views/course_outline.py

```diff
@@ -154,3 +160,18 @@ def mark_first_unit_to_resume(self, block_node):
         if children:
             children[0]['resume_block'] = True
             self.mark_first_unit_to_resume(children[0])
+
+
+@ensure_csrf_cookie
+def reset_course_deadlines(request, course_id):
+    """
+    Set the start_date of a schedule to today, which in turn will adjust due dates for
+    sequentials belonging to a self paced course
+    """
+    course = CourseOverview.objects.get(id=course_id)
+    if course.self_paced:
+        enrollment = CourseEnrollment.objects.get(user=request.user, course=course_id)
+        schedule = enrollment.schedule
```

Comment:
Are we guaranteed that the enrollment will have a schedule? If not, the next line of calling schedule.start_date could fail with a NoneType has no attribute start_date